### ISSUE TITLE
Drop rows from eset if they have no GPL annotation

### ISSUE THREAD

**Issue by russHyde** [russHyde/miiq#10]
in `add_gpl_to_eset` when a rowname of the eset has no corresponding entry in the GPL dataset, remove that row from the eset and warn the user that some rows were omitted

**Comment by russHyde** [comment]
Note that when annotating GSE71869 with GPL17692, the id 17000000 was absent from the GPL dataset

**Comment by russHyde** [comment]
Need:
- A test where we make an ExpressionSet and a [GPL dataset? - TODO check the structure]
    - [x] - Check that the Eset is returned with unchanged row order when all IDs are present in the GPL
    - [x] - Check that the Eset is unchanged row order (except for removing missing IDs) when some eset IDs are absent from the GPL
    - [x] - Check that a warning is thrown when a smaller Eset is returned than is input

**Comment by russHyde** [comment]
Current code:
~~~~
miiq:::add_gpl_to_eset
function (eset, gpl_data) 
{
    stopifnot(methods::is(eset, "ExpressionSet"))
    stopifnot(methods::is(gpl_data, "GPL"))
    gpl_table <- GEOquery::Table(gpl_data)
    stopifnot("ID" %in% colnames(gpl_table))
    if (!all(rownames(eset) %in% gpl_table[["ID"]])) {
        stop("IDs in `eset` but not `gpl_table`: ", utils::head(setdiff(rownames(eset), 
            gpl_table[["ID"]])))
    }
    annot_ord <- match(rownames(eset), gpl_table[["ID"]])
    fd <- new("AnnotatedDataFrame", data = gpl_table[annot_ord, 
        ], varMetadata = GEOquery::Columns(gpl_data))
    rownames(fd) <- rownames(eset)
    Biobase::featureData(eset) <- fd
    eset
}
~~~~

**Comment by russHyde** [comment]
GPL objects contain:
- dataTable: GEODataTable
- header: list

~~~~
my_gpl <- new("GPL", header = list(), dataTable = new("GEODataTable"))
~~~~

**Comment by russHyde** [comment]
GEODataTable objects contain:
- columns: data.frame
    - Two columns: "Column", "Description"
    - Entries in "Column" match the colnames in 'table'
- table: data.frame
    - An `ID` column is present in GPL-derived `GEODataTable`s

So we can make a test GPL dataset using:

~~~~
test_gpl <- new(
    "GPL",
    header = list(),
    dataTable = new(
        "GEODataTable",
        table = data.frame(
            ID = paste0("G", c(1, 2, 4)),
            annot = c("x", "y", "z"),
            stringsAsFactors = FALSE
        ),
        columns = data.frame(
            Column = c("ID", "annot"),
            Description = c("Unique identifier for the probe", "Some annotation"),
            stringsAsFactors = FALSE
        )
    )
)
~~~~

**Comment by russHyde** [comment]
fixed in #11
